docs/DeveloperGuide.md

```diff
@@ -0,0 +1,128 @@
+# Developer Guide
+Thank you for contributing to RepoSense!
+- [Setting Up](#setting-up)
+- [Architecture](#architecture)
+- [Notable POJOs](#notable-pojos)
+- [Frontend Components](#frontend-components)
+- [Dashboard Generator Components](#dashboard-generator-components)
+- [Git Utilities](#git-utilities)
+- [Analyzer Components](#analyzer-components)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+
+1. **JDK `1.8.0_60`**  or later
+2. **Git** on the command line 
+3. **findstr** for Windows, **grep** for macOS or Linux on the command line
+   * Ensure that you're able to use these tools on the OS terminal.
+4. **IntelliJ** IDE [Optional]
+
+### Setting up the project in your computer using IntelliJ
+1. Fork this repo, and clone the fork to your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle.
+4. Click `Configure` > `Project Defaults` > `Project Structure`.
+5. Click `New…​` and find the directory of the JDK.
+6. Click `Import Project`.
+7. Locate the build.gradle file and select it. Click OK.
+8. Ensure that the selected version of `Gradle JVM` matches our prerequisite.
+9. Click `OK` to accept the all the other default settings.
+
+#### Verifying the setup
+1. Ensure that Gradle build without error.
+2. Run the tests to ensure they all pass.
+   1. Go to `src` -> `main` -> `test`. Right click -> Run Tests.
```

Comment:
Let's change this to be independent of IDE; using gradle to run tests. Also include functional test.